### ISSUE TITLE
Server Side renderer always renders login screen

### ISSUE THREAD

**Issue by flyersa** [grafana/grafana#6766]
Hi,

im using the Grafana 4.0 version and setup alerting including webdav upload for the panels. However in the emails from the alerts and on the webdav share all i see is always the grafana login screen as image. Is there something i miss for configuration to have the server side renderer do authentication to local grafana in order to take the picture?

we use no http proxy in front of grafana, root_url is set correctly, we use ldap authentication. I also tried to hack into an api key or credentials into render.js all without success :(

**Comment by flyersa** [comment]
its the same when i use:

http://grafana:3000/render/dashboard-solo/db/mysql-cluster?from=1480517657073&to=1480539257073&panelId=5&width=1000&height=500

as example. even when im logged in, then i always get presented with a image of the login screen. If i adjust render.js to send a authorisation bearer then it returns 404 not found. I dont get what is wrong. Also tried with Anonymous Viewer, always login screen.

**Comment by fdelapena** [comment]
Same here, It was working with 3.1.1, became broken since the 4.0.0 update. Same settings as described in #5906 for Apache proxy, not needing proxy auth.
I was using `http` in the `root_url` before and with trailing slash, tried with `https` and without trailing but with the same result:

I'm logged and something like this works:

    https://grafana.example.com/dashboard/db/whatever

However in a new tab trying this shows a rendered PNG login:

    https://grafana.example.com/render/dashboard/db/whatever

`grafana.log` says this:

```
t=2016-11-30T15:08:59-0600 lvl=info msg=Rendering logger=png-renderer path=dashboard/db/whatever?
t=2016-11-30T15:09:00-0600 lvl=info msg="Request Completed" logger=context userId=0 orgId=0 uname= method=GET path=/dashboard/db/whatever status=302 remote_addr=127.0.0.1 time_ms=0s size=29
```

**Comment by fdelapena** [comment]
In my case, the issue got worked around by resetting to default the `http_addr` (0.0.0.0), I was using 127.0.0.1 to prevent exposing the port 3000 as it was being handled via proxy.

**Comment by kylejvrsa** [comment]
I found that not enabling enforce_domain, as in keeping it at it's default also allows you to see the graphs in the email again.

# Redirect to correct domain if host header does not match domain
   # Prevents DNS rebinding attacks
   ;enforce_domain = false

**Comment by flyersa** [comment]
none of that works for me, not setting bind to 0.0.0.0 and also the enforce_domain does not help at all, also as logged in user i cant see anything except the login screen, same if i click the "Share" button on a panel and click on the "Direct link rendered image" .

**Comment by flyersa** [comment]
well i fixed it by adjusting render.js to remove the /render/ part from the url and in addition to add a authorisation key.


 page.customHeaders = {
  "Authorization": "Bearer xxxxx"
 };

params.url.replace("render", "");

not clean but it works for us now.

**Comment by torkelo** [comment]
@flyersa strange unable to replicate any issues, the rendering should not require any auth, grafana will do a local call to it's self via phantomjs, passing a single use token to phantom

**Comment by flyersa** [comment]
@torkelo  i think its the same then this as the behavior is the same:

https://github.com/grafana/grafana/issues/6769

**Comment by torkelo** [comment]
Did you have enforce domain enabled? Thought you said it made no difference?

**Comment by silvestre-abruzzo** [comment]
Hi I came across to the same issue yesterday. Here I propose a possible solution https://github.com/grafana/grafana/pull/6813

**Comment by hvnsweeting** [comment]
I got this same problem after upgraded to 4.0

This is command from ps output when it rendering

```
/usr/share/grafana/vendor/phantomjs/phantomjs --ignore-ssl-errors=true /usr/share/grafana/vendor/phantomjs/render.js url='http://localhost:3000/dashboard-solo/db/infra?from=1480845231174&to=1480931631174&var-server=jenkins-slave-go-1&panelId=1&width=1000&height=500' width=1000 height=500 png=/var/lib/grafana/png/WiE3WkRWtaDDge8F8SPO.png domain=metrics.XXX renderKey=X62UTDqBam6qs2zftCF3DjZ7Hp5Tan3U
```

I also changed render.js to set passed domain as in #6813 

My config

```
protocol = http
http_addr = localhost
http_port = 3000

domain = metrics.XXX
root_url = https://metrics.XXX
enforce_domain = false
```

**Comment by torkelo** [comment]
This has been fixed in master and v4.0.x branch, try latest nightly build
http://grafana.org/builds/

**Comment by hvnsweeting** [comment]
I tried latest nightly build and it rendered now, with my config keeps still as shown above. Thank you very much.
